### PR TITLE
#5615 Fix: change signature when switching different send mail address

### LINKED ISSUE #5615: email signature does not change when switching between different "send mail as identities"
Hello,

we are using latest 8.5.3 on Firefox and we have some Gmail accounts with multiple "Identities" which use their own email signature (under Settings -> Account -> "Send mail as" we have multiple email addresses and they are configured to each use a specific email signature).

Switching between different "from" addresses in the normal Gmail composer updates the email signature in the email body. 

However, this is not happening when using secure compose: I can switch the "from" address, but the signature will stay the same (see screenshot)
image

 I found https://github.com/FlowCrypt/flowcrypt-browser/issues/61 but I'm not sure if this is the same issue ?

kind regards,

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented a fix to change signature when switching different send mail address.

close #5615 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/chrome/elements/compose-modules/compose-sender-module.ts [hunk 1]
```diff
@@ -67,5 +67,9 @@ export class ComposeSenderModule extends ViewModule<ComposeView> {
   private actionInputFromChangeHandler = async () => {
     await this.view.recipientsModule.reEvaluateRecipients(this.view.recipientsModule.getValidRecipients());
     this.view.footerModule.onFooterUpdated(await this.view.footerModule.getFooterFromStorage(this.view.senderModule.getSender()));
+    const signaturePattern = /--\s*<br\s*\/?>[\s\S]*$/i;
```
Comment (sosnovsky) [inline extension/chrome/elements/compose-modules/compose-sender-module.ts:70]: this regex works well for new messages, but when you compose reply message and switch send mail address - it removes older messages content too.

need to update this regex to leave older messages text (starting from `On ... wrote:...`)

Comment (ioanmo226) [reply extension/chrome/elements/compose-modules/compose-sender-module.ts]: Nice catch :+1: 
Implemented a fix and pushed

**Review by sosnovsky — COMMENTED** [review]
It works better now, but doesn't persist new line between signature and older messages content:

Screenshot 2024-08-21 at 12 06 05

**Comment by ioanmo226** [comment]
Oops!. Nice catch :+1:
Tested with reply/forward/secure compose but didn't try with draft.
Let me fix

**Review by sosnovsky — APPROVED** [review]
All good now, thanks!

**Pull request merged**